Fix #30.

jdmichaud opened the pull request:
* C99 specifies that `__STDC_FORMAT_MACROS` must be defined for `PRIu64` to be defined.
* Make RthinkDB::Custor a full fledged std::iterator.

jdmichaud committed:
Fix #30.

jdmichaud committed:
Make RethinkDB::Cursor a valid std::iterator.

AtnNn commented:
The first commit LGTM, thanks!

The second commit however looks wrong. I'm pretty sure it's not a forward iterator. The `value_type` is Datum, not Cursor. And I don't know if `difference_type` makes any sense here, but I don't think it would be `ptrdiff_t`.

Commit:
Cursor::iterator is an input_iterator.
Cursor::iterator::value_type is a Datum.
Remove Cursor::iterator::difference_type.

mbroadst commented:
@jdmichaud I wonder if you could possibly commit a unit test that expresses the need for inheriting from `std::iterator` ? From some quick searches it looks like there's interest in deprecating `std::iterator` in the next draft (http://stackoverflow.com/questions/6471019/can-should-i-inherit-from-stl-iterator).
